### PR TITLE
Prepare for release 0.4.1.

### PR DESCRIPTION
**Pull request by simon-shopify**
There were a bunch of changes sitting on `master`. Let's release them?

### CODE REVIEW

File: lib/identity_cache.rb [hunk 1]
```diff
@@ -59,13 +59,13 @@ class << self
...
     mattr_accessor :never_set_inverse_association
-    self.never_set_inverse_association = version >= Gem::Version.new("0.5")
+    self.never_set_inverse_association = version >= Gem::Version.new("0.6")
 
     # Fetched records are not read-only and this could sometimes prevent IDC from
     # reflecting what's truly in the database when fetch_read_only_records is false.
     # When set to true, it will only return read-only records when cache is used.
     mattr_accessor :fetch_read_only_records
-    self.fetch_read_only_records = version >= Gem::Version.new("0.5")
+    self.fetch_read_only_records = version >= Gem::Version.new("0.6")
```
Comment (casperisfine) [inline lib/identity_cache.rb]: That's super weird. Who wrote that?

Comment (dylanahsmith) [inline lib/identity_cache.rb]: these are breaking changes that we want to make for 0.5.  Don't just bump the number.  They are configurations so we can always use the old behaviour in Shopify until we can actually turn these on.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 # IdentityCache changelog
 
+#### 0.5.0
+
+- Deprecated embedded associations on models that don't use IDC (#305)
+- Remove a respond_to? check that hides mistakes in includes hash (#307)
+- refactor: Include modules into IdentityCache instead into models (#303)
```
Comment (dylanahsmith) [inline CHANGELOG.md]: Refactoring shouldn't matter to the user's of the library.  Only include changes that affect the users of the library

**Comment by dylanahsmith** [comment]
Are there any changes that necessitate a 0.5 release rather than a 0.4.1 release?

File: lib/identity_cache.rb (hunk 1 above)
Comment (casperisfine) [inline lib/identity_cache.rb]: Right, but then now that we are shipping 0.6? we can simply resolve that to `true`

**Comment by simon-shopify** [comment]
> Are there any changes that necessitate a 0.5 release rather than a 0.4.1 release?

Yeah I don't think there are any breaking changes, I can make it 0.4.1.

**Pull request merged**